### ISSUE TITLE
onHostDestroy getCurrentActivity is already null, prevents unbinding from services

### ISSUE THREAD

**Issue by mauron85** [facebook/react-native#9791]
### Issue Description

Reference to current activity is null in lifecycle event handler **onHostDestroy**. This prevents from proper unbinding from any kind of Android service and results in:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'void android.app.Activity.unbindService(android.content.ServiceConnection)' on a null object reference
```

### Steps to Reproduce / Code Snippets

```
  @Override
  public void onHostDestroy() {
    final Activity currentActivity = this.getCurrentActivity(); //return null
    currentActivity.unbindService(mConnection); // null pointer instead activity will crash app
  }
```

#### Expected Results

this.getCurrentActivity() returns instance of Activity

### Additional Information

* React Native version: 0.29 but believe it will apear also in recent versions
* Platform(s) (iOS, Android, or both?): Android
* Operating System (macOS, Linux, or Windows?): all

**Comment by fab1an** [comment]
Please try this with 0.33.0-rc1.

**Comment by mauron85** [comment]
@fab1an fixed in 0.33.0-rc1. Many thanks.
